pkg/client/event.go

```diff
@@ -14,14 +14,24 @@ import (
 
 type PushOpFunc func(*eventcontracts.PushEventRequest) error
 
+type BulkPushOpFunc func(*eventcontracts.BulkPushEventRequest) error
+
 type EventClient interface {
 	Push(ctx context.Context, eventKey string, payload interface{}, options ...PushOpFunc) error
 
+	BulkPush(ctx context.Context, payloads []EventWithMetadata, options ...BulkPushOpFunc) error
+
 	PutLog(ctx context.Context, stepRunId, msg string) error
 
 	PutStreamEvent(ctx context.Context, stepRunId string, message []byte) error
 }
 
+type EventWithMetadata struct {
```

Comment:
not pretty, but `AdditionalMetadata` is named as such because of a conflict on all our API resources where `metadata` is reserved for `createdAt`, `updatedAt`, and `ID`s...